src/Oden/Core/Definition.hs

```diff
@@ -16,3 +16,11 @@ data Definition e
   | ProtocolDefinition (Metadata SourceInfo) QualifiedName Protocol
   | Implementation (Metadata SourceInfo) (ProtocolImplementation e)
   deriving (Show, Eq, Ord)
+
+definitionSourceInfo :: Definition e -> SourceInfo
+definitionSourceInfo (Definition (Metadata si) _ _)          = si
+definitionSourceInfo (ForeignDefinition (Metadata si) _ _)   = si
+definitionSourceInfo (TypeDefinition (Metadata si) _ _ _)    = si
+definitionSourceInfo (ProtocolDefinition (Metadata si) _ _)  = si
+definitionSourceInfo (Implementation (Metadata si) _)        = si
+
```

Comment:
There's a type class called `HasSourceInfo` (https://github.com/d-dorazio/oden/blob/8c1f160a9526df39e877df6ccb01487b698e1606/src/Oden/SourceInfo.hs#L25) that should work here, instead of having the on-off function `definitionSourceInfo`.